test/jit/test_graph_rewrite_passes.py

```diff
@@ -4,58 +4,131 @@
 import torch
 import torch._C
 from torch.testing import FileCheck
+from typing import Callable
 
+class FunctionalConv2d(torch.nn.Module):
+    def __init__(self, weight: torch.Tensor, bias: torch.Tensor):
+        super(FunctionalConv2d, self).__init__()
+        self.weight = weight
+        self.bias = bias
+
+    def forward(self, x: torch.Tensor):
+        x = torch.nn.functional.conv2d(
+            input=x.unsqueeze(dim=0),
+            weight=self.weight,
+            bias=self.bias,
+        )
+        return x
+
+class FunctionalLinear(torch.nn.Module):
+    def __init__(self, weight: torch.Tensor, bias: torch.Tensor = None):
+        super(FunctionalLinear, self).__init__()
+        self.weight = weight
+        self.bias = bias
+
+    def forward(self, x: torch.Tensor):
+        res = torch.matmul(x, self.weight.t())
+        if self.bias is not None:
+            res.add_(self.bias)
+        return res
+
+class FunctionalMatmul(torch.nn.Module):
+    def __init__(self, weight: torch.Tensor):
+        super(FunctionalMatmul, self).__init__()
+        self.weight = weight
+
+    def forward(self, x: torch.Tensor):
+        return torch.matmul(x, self.weight)
 
 class TestGraphRewritePasses(JitTestCase):
-    def test_fuse_linear(self):
-        class FunctionalLinear(torch.nn.Module):
-            def __init__(self, weight, bias):
-                super(FunctionalLinear, self).__init__()
-                self.weight = weight
-                self.bias = bias
-
-            def forward(self, x):
-                res = torch.matmul(x, self.weight.t())
-                if self.bias is not None:
-                    res.add_(self.bias)
-                return res
+    def check_single_replacement(
+        self,
+        old_kind: str,
+        new_kind: str,
+        jit_pass: Callable[[str], None],
+        model: torch.jit.ScriptModule
+    ):
+        for node in model.graph.nodes():
+            if node.kind() == old_kind:
+                source_range_1 = node.sourceRange()
+        jit_pass(model.graph)
+        for node in model.graph.nodes():
+            if node.kind() == new_kind:
+                source_range_2 = node.sourceRange()
+        self.assertTrue(source_range_1 == source_range_2)
+
+    def check_op_presence(
+        self,
+        check_yes: list[str],
+        check_not: list[str],
+        jit_pass: Callable[[str], None],
+        model: torch.jit.ScriptModule
+    ):
+        jit_pass(model.graph)
+        for cn in check_not:
+            FileCheck().check_not(cn).run(model.graph)
```

Comment:
Have you checked if there a way to do all check/check_not in one go? I think when I did this last in https://github.com/pytorch/pytorch/blob/master/test/test_xnnpack_integration.py#L611, I dont think I optimized it. You can probably combine all with like here: https://github.com/pytorch/pytorch/blob/master/test/test_mobile_optimizer.py#L114-L123